test/integration/admin/partner_integration_test.rb

```diff
@@ -115,6 +115,47 @@ class PartnerIntegrationTest < ActionDispatch::IntegrationTest
     tag = tag_options.first
     assert tag.attributes.key?('selected')
   end
+
+  test 'Partner create form gives feedback on bad image selection' do
+    new_partner_params = {
+      name: 'A partner',
+      address_attributes: {
+        street_address: @partner.address.street_address,
+        postcode: @partner.address.postcode
+      },
+      image: fixture_file_upload("bad-cat-picture.bmp")
+    }
+
+    sign_in @admin
+    post admin_partners_path, params: { partner: new_partner_params }
+
+    assert_not response.redirect?
+
+    assert_select "h6", text: "1 error prohibited this Partner from being saved"
+    assert_select '#form-errors li', text: "Image You are not allowed to upload \"bmp\" files, allowed types: jpg, jpeg, gif, png"
+    assert_select 'form .partner_image .invalid-feedback', text: "Image You are not allowed to upload \"bmp\" files, allowed types: jpg, jpeg, gif, png"
```

Comment:
I haven't looked, but the inclusion of both `jpg` and `jpeg` here makes me wonder: are we just checking the file extension or are we actually looking at the contents of the file to work out what it is?

Reply:
I don't know as this is all done with carrierwave before my time on the code. I think we need to go through all the uploaders for site, partner, article and user and consolidate the acceptable formats. (frankly i would only allow jpgs as they are smaller in size and pretty well known in none-tech circles but that is just me)

Reply:
pretty sure we are just looking at the extension yeah. and no we should allow pngs, gifs and wepbs at least :)

Reply:
If we care about size, we can post-process images into webp with jpg fallbacks and accept any format. But you know, later.

> pretty sure we are just looking at the extension

This error message could be better probably, but one for a future iteration.

Reply:
I will make this into a chore ticket